internal/helpers/database/database.go

```diff
@@ -33,11 +33,17 @@ import (
 func FetchDatabaseData(ctx context.Context, dbcr *kindav1beta1.Database, dbCred database.Credentials, instance *kindav1beta1.DbInstance) (database.Database, *database.DatabaseUser, error) {
 	log := log.FromContext(ctx)
 	host := instance.Status.Info["DB_CONN"]
-	port, err := strconv.Atoi(instance.Status.Info["DB_PORT"])
+	port64, err := strconv.ParseUint(instance.Status.Info["DB_PORT"], 10, 16)
 	if err != nil {
 		log.Error(err, "can't get port information from the instanceRef")
 		return nil, nil, err
 	}
+	if port64 < 0 || port64 > 65535 {
```

Comment:
1. uint64 min value is 0, so you can't remove the first check
2. the max value of uint16 should be 65535, so maybe makes sense to use it instead?